### DISCUSSION TITLE
How fix "Error: Body exceeded 1mb limit" when i uses  api actions?

### DISCUSSION THREAD

**Discussion by baldTooth** [vercel/next.js#51097] [App Router] [answered]
How fix "Error: Body exceeded 1mb limit" when i uses  api actions? 

Thanks/

**Comment by devjiwonchoi** [comment, marked as answer]
Increase the `response size limit` by using a custom config in your `API` file.

```js
export const config = {
  api: {
    responseLimit: false | '8mb' | YOUR_DESIRED_SIZE_LIMIT,
  },
}
```

## UPDATE

This feature has been blocked since `App Router`. It is currently hard-coded to `1mb` as you can see [here](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/action-handler.ts#L355).

You need to customize the `parseBody()` size manually through `node_modules`.. not recommended but no "quicker" option is available at this moment.

The file is located in `node_modules\next\dist\server\app-render\action-handler.js`.

![Screenshot 2023-06-11 at 12 55 49 AM](https://github.com/vercel/next.js/assets/120007119/0447377a-31b3-4934-9856-f59aeca36c58)

## UPDATE2

Resolved with #51104

## Reference
- [API Routes Response Size Limit](https://nextjs.org/docs/messages/api-routes-response-size-limit)
- [Action Handler Set to `1mb`](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/action-handler.ts#L355)

**Reply by baldTooth** [reply]
A lot of  thanks!!)

**Reply by devjiwonchoi** [reply]
@baldTooth `serverActiionsBodySizeLimit` has been added on [release v13.4.8](https://github.com/vercel/next.js/releases/tag/v13.4.8).
